crates/bindings-sys/src/lib.rs

```diff
@@ -332,11 +332,58 @@ pub mod raw {
         /// - `NOT_IN_TRANSACTION`, when called outside of a transaction.
         /// - `NO_SUCH_TABLE`, when `table_id` is not a known ID of a table.
         /// - `BSATN_DECODE_ERROR`, when `row` cannot be decoded to a `ProductValue`.
-        ///   typed at the `ProductType` the table's schema specifies.
+        ///    typed at the `ProductType` the table's schema specifies.
         /// - `UNIQUE_ALREADY_EXISTS`, when inserting `row` would violate a unique constraint.
         /// - `SCHEDULE_AT_DELAY_TOO_LONG`, when the delay specified in the row was too long.
         pub fn datastore_insert_bsatn(table_id: TableId, row_ptr: *mut u8, row_len_ptr: *mut usize) -> u16;
 
+        /// Updates a row in the table identified by `table_id` to `row`
+        /// where the row is read from the byte string `row = row_ptr[..row_len]` in WASM memory
+        /// where `row_len = row_len_ptr[..size_of::<usize>()]` stores the capacity of `row`.
+        ///
+        /// The byte string `row` must be a BSATN-encoded `ProductValue`
+        /// typed at the table's `ProductType` row-schema.
+        ///
+        /// The row to update is found by projecting `row`
+        /// to the type of the *unique* index identified by `index_id`.
+        /// If no row is found, `row` is inserted.
```

Comment:
> If no row is found, `row` is inserted.

I do not agree with this choice. My intuition is that an `update` call where no such row previously existed is more likely to be a bug than intended behavior. As such, I think an `update` that finds no such row should roll back the insertion and return an error. I don't see this being a performance concern, as we expect that branch to be rarely taken.